Importamos las librerias que vamos a utilizar.

In [1]:
import pandas as pd
import numpy as np

In [2]:
# Establecer la opción para mostrar todas las columnas
pd.set_option('display.max_columns', None)


# Carga de datos.

In [3]:
movies_raw = pd.read_csv('data/movies_dataset.csv',low_memory=False)

-----

# Analisis de movies_raw.

In [4]:
movies_raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45466 entries, 0 to 45465
Data columns (total 24 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   adult                  45466 non-null  object 
 1   belongs_to_collection  4494 non-null   object 
 2   budget                 45466 non-null  object 
 3   genres                 45466 non-null  object 
 4   homepage               7782 non-null   object 
 5   id                     45466 non-null  object 
 6   imdb_id                45449 non-null  object 
 7   original_language      45455 non-null  object 
 8   original_title         45466 non-null  object 
 9   overview               44512 non-null  object 
 10  popularity             45461 non-null  object 
 11  poster_path            45080 non-null  object 
 12  production_companies   45463 non-null  object 
 13  production_countries   45463 non-null  object 
 14  release_date           45379 non-null  object 
 15  re

Tenemos 45466 filas y 24 columnas. 
Algunas columnas presentan inconsistencias en sus datos y valores nulos.

In [5]:
movies_raw.describe(include='all')

adult                              belongs_to_collection budget  \
count   45466                                               4494  45466   
unique      5                                               1698   1226   
top     False  {'id': 415931, 'name': 'The Bowery Boys', 'pos...      0   
freq    45454                                                 29  36573   
mean      NaN                                                NaN    NaN   
std       NaN                                                NaN    NaN   
min       NaN                                                NaN    NaN   
25%       NaN                                                NaN    NaN   
50%       NaN                                                NaN    NaN   
75%       NaN                                                NaN    NaN   
max       NaN                                                NaN    NaN   

                               genres                     homepage      id  \
count                           45466                         7782   45466   
unique                           4069                         7673   45436   
top     [{'id': 18, 'name': 'Drama'}]  http://www.georgecarlin.com  141971   
freq                             5000                           12       3   
mean                              NaN                          NaN     NaN   
std                               NaN                          NaN     NaN   
min                               NaN                          NaN     NaN   
25%                               NaN                          NaN     NaN   
50%                               NaN                          NaN     NaN   
75%                               NaN                          NaN     NaN   
max                               NaN                          NaN     NaN   

          imdb_id original_language original_title            overview  \
count       45449             45455          45466               44512   
unique      45417                92          43373               44307   
top     tt1180333                en         Hamlet  No overview found.   
freq            3             32269              8                 133   
mean          NaN               NaN            NaN                 NaN   
std           NaN               NaN            NaN                 NaN   
min           NaN               NaN            NaN                 NaN   
25%           NaN               NaN            NaN                 NaN   
50%           NaN               NaN            NaN                 NaN   
75%           NaN               NaN            NaN                 NaN   
max           NaN               NaN            NaN                 NaN   

       popularity                       poster_path production_companies  \
count       45461                             45080                45463   
unique      43758                             45024                22708   
top           0.0  /5D7UBSEgdyONE6Lql6xS7s6OLcW.jpg                   []   
freq           66                                 5                11875   
mean          NaN                               NaN                  NaN   
std           NaN                               NaN                  NaN   
min           NaN                               NaN                  NaN   
25%           NaN                               NaN                  NaN   
50%           NaN                               NaN                  NaN   
75%           NaN                               NaN                  NaN   
max           NaN                               NaN                  NaN   

                                     production_countries release_date  \
count                                               45463        45379   
unique                                               2393        17336   
top     [{'iso_3166_1': 'US', 'name': 'United States o...   2008-01-01   
freq                                                17851          136  

In [6]:
movies_raw.head()

adult                              belongs_to_collection    budget  \
0  False  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1  False                                                NaN  65000000   
2  False  {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   
3  False                                                NaN  16000000   
4  False  {'id': 96871, 'name': 'Father of the Bride Col...         0   

                                              genres  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   
3  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
4                     [{'id': 35, 'name': 'Comedy'}]   

                               homepage     id    imdb_id original_language  \
0  http://toystory.disney.com/toy-story    862  tt0114709                en   
1                                   NaN   8844  tt0113497                en   
2                                   NaN  15602  tt0113228                en   
3                                   NaN  31357  tt0114885                en   
4                                   NaN  11862  tt0113041                en   

                original_title  \
0                    Toy Story   
1                      Jumanji   
2             Grumpier Old Men   
3            Waiting to Exhale   
4  Father of the Bride Part II   

                                            overview popularity  \
0  Led by Woody, Andy's toys live happily in his ...  21.946943   
1  When siblings Judy and Peter discover an encha...  17.015539   
2  A family wedding reignites the ancient feud be...    11.7129   
3  Cheated on, mistreated and stepped on, the wom...   3.859495   
4  Just when George Banks has recovered from his ...   8.387519   

                        poster_path  \
0  /rhIRbceoE9lR4veEXuwCC2wARtG.jpg   
1  /vzmL6fP7aPKNKPRTFnZmiUfciyV.jpg   
2  /6ksm1sjKMFLbO7UY2i6G1ju9SML.jpg   
3  /16XOMpEaLWkrcPqSQqhTmeJuqQl.jpg   
4  /e64sOI48hQXyru7naBFyssKFxVd.jpg   

                                production_companies  \
0     [{'name': 'Pixar Animation Studios', 'id': 3}]   
1  [{'name': 'TriStar Pictures', 'id': 559}, {'na...   
2  [{'name': 'Warner Bros.', 'id': 6194}, {'name'...   
3  [{'name': 'Twentieth Century Fox Film Corporat...   
4  [{'name': 'Sandollar Productions', 'id': 5842}...   

                                production_countries release_date  \
0  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-10-30   
1  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-15   
2  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22   
3  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22   
4  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-02-10   

       revenue  runtime                                   spoken_languages  \
0  373554033.0     81.0           [{'iso_639_1': 'en', 'name': 'English'}]   
1  262797249.0    104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   
2          0.0    101.0           [{'iso_639_1': 'en', 'name': 'English'}]   
3   81452156.0    127.0           [{'iso_639_1': 'en', 'name': 'English'}]   
4   76578911.0    106.0           [{'iso_639_1': 'en', 'name': 'English'}]   

     status                                            tagline  \
0  Released                                                NaN   
1  Released          Roll the dice and unleash the excitement!   
2  Released  Still Yelling. Still Fighting. Still Ready for...   
3  Released  Friends are the people who let you be yourself...   
4  Released  Just When His World Is Back To Normal... He's ...   

                         title  video  vote_average  vote_count  
0                    Toy Story  False           7.7      5415.0  
1                      Jumanji  False           6.9      2413.0  
2             Grumpier Old Men  False           6.5        92.0  
3            Waiting to Exh

Podemos observar 5 columnas que contienen datos anidados. Estas son: 'belongs_to_collection', 'genres', 'production_companies', 'production_countries' y 'spoken_languages'.

----


# Duplicados

¿Hay filas duplicadas? ¿Cuántas?


In [7]:
#filas duplicadas
(
    movies_raw
    .duplicated(keep=False)
    .sum()
)

33

In [8]:
# Filas con campo id duplicado /  Una misma pelicula aparece varias veces en el dataframe.
(
    movies_raw
    .duplicated(subset='id',keep=False)
    .sum()
)

59

Eliminaremos las filas duplicadas y las que poseen ids duplicados. 
Ya que representan menos del 1% del dataset.
Pero se consultará para posteriores entregas.


---

# Datos nulos

¿Cuántos datos nulos tiene cada columna?

In [9]:
(
    movies_raw
    .isnull()
    .sum()
)

adult                        0
belongs_to_collection    40972
budget                       0
genres                       0
homepage                 37684
id                           0
imdb_id                     17
original_language           11
original_title               0
overview                   954
popularity                   5
poster_path                386
production_companies         3
production_countries         3
release_date                87
revenue                      6
runtime                    263
spoken_languages             6
status                      87
tagline                  25054
title                        6
video                        6
vote_average                 6
vote_count                   6
dtype: int64

¿Cuáles son las 6 peliculas que no tienen título especificado?

In [10]:
movies_raw[movies_raw['title'].isnull()]

adult  \
19729                                              False   
19730                                 - Written by Ørnås   
29502                                              False   
29503   Rune Balot goes to a casino connected to the ...   
35586                                              False   
35587   Avalanche Sharks tells the story of a bikini ...   

                                   belongs_to_collection  \
19729                                                NaN   
19730                                           0.065736   
29502  {'id': 122661, 'name': 'Mardock Scramble Colle...   
29503                                           1.931659   
35586                                                NaN   
35587                                           2.185485   

                                 budget  \
19729                                 0   
19730  /ff9qCepilowshEtG2GYWwzt2bs4.jpg   
29502                                 0   
29503  /zV8bHuSL6WXoD6FWogP9j4x80bL.jpg   
35586                                 0   
35587  /zaSf5OG7V8X8gqFvly88zDdRm46.jpg   

                                                  genres  \
19729  [{'id': 28, 'name': 'Action'}, {'id': 53, 'nam...   
19730  [{'name': 'Carousel Productions', 'id': 11176}...   
29502  [{'id': 16, 'name': 'Animation'}, {'id': 878, ...   
29503  [{'name': 'Aniplex', 'id': 2883}, {'name': 'Go...   
35586  [{'id': 10770, 'name': 'TV Movie'}, {'id': 28,...   
35587  [{'name': 'Odyssey Media', 'id': 17161}, {'nam...   

                                                homepage          id  \
19729                                                NaN       82663   
19730  [{'iso_3166_1': 'CA', 'name': 'Canada'}, {'iso...  1997-08-20   
29502                      http://m-scramble.jp/exhaust/      122662   
29503  [{'iso_3166_1': 'US', 'name': 'United States o...  2012-09-29   
35586                                                NaN      249260   
35587           [{'iso_3166_1': 'CA', 'name': 'Canada'}]  2014-01-01   

         imdb_id original_language                            original_title  \
19729  tt0113002                en                              Midnight Man   
19730          0             104.0  [{'iso_639_1': 'en', 'name': 'English'}]   
29502  tt2423504                ja                          マルドゥック・スクランブル 排気   
29503          0              68.0      [{'iso_639_1': 'ja', 'name': '日本語'}]   
35586  tt2622826                en                          Avalanche Sharks   
35587          0              82.0  [{'iso_639_1': 'en', 'name': 'English'}]   

                                                overview  \
19729  British soldiers force a recently captured IRA...   
19730                                           Released   
29502         Third film of the Mardock Scramble series.   
29503                                           Released   
35586  A group of skiers are terrorized during spring...   
35587                                           Released   

                  popularity                          poster_path  \
19729                    NaN                                  NaN   
19730                    NaN                         Midnight Man   
29502                    NaN                                  NaN   
29503                    NaN  Mardock Scramble: The Third Exhaust   
35586                    NaN                                  NaN   
35587  Beware Of Frost Bites                     Avalanche Sharks   

      production_companies production_countries release_date  revenue  \
19729                  NaN                  NaN          NaN      NaN   
19730                False                  6.0            1      NaN   
29502                  NaN                  NaN          NaN      NaN   
29503                False                  7.0           12      NaN   
35586                  NaN                  NaN          NaN      NaN   
35587                False                  4.3           22      NaN   

   

Estas 6 filas también se eliminaran ya que la mayoría de sus datos son nulos o inconsistentes.

----